src/QirRuntime/test/QIR-static/qsharp/qir-test-arrays.qs

```diff
@@ -40,6 +41,8 @@ namespace Microsoft.Quantum.Testing.QIR {
             let res5 = LogTest();
             let res6 = ArcTan2Test();
             let res7 = PauliToStringTest();
+            let res8 = TestDrawRandomInt(0, 1);
```

Comment:
Out of curiosity, why are the math and random tests added to the `qir-test-arrays.qs` file? It feels a bit surprising for an operation named `Test_Arrays` to test things unrelated to arrays.

Reply:
This is a trick to cheat the compiler.

We are building the _executable_ from the .qs files. And the compiler includes into the executable only _those functions_, that _are reachable_ from the `@Entrypoint` function. 
```
    @EntryPoint()
    operation Test_Arrays(array : Int[], index : Int, val : Int, compilerDecoy : Bool) : Int {
```
And for the `@Entrypoint` function the compiler generates a special wrapper that can be called from our C++ test code, see in `qir-gen.ll`
```
define i64 @Microsoft__Quantum__Testing__QIR__Test_Arrays(i64 %array__count, i64* %array, i64 %index, i64 %val, i1 %compilerDecoy) #0 {
```
Pay attention to the first 2 parameters as opposed to the 1st parameter in the Q# declaration for the `@Entrypoint` function shown earlier. And compare to the 1st parameter in the actual implementation of the `@Entrypoint` function in `qir-gen.ll`:
```
define i64 @Microsoft__Quantum__Testing__QIR__Test_Arrays__body(%Array* %array, i64 %index, i64 %val, i1 %compilerDecoy) {
```

The other option would be for the compiler to generate the _library_, rather than _executable_. In that case _all_ the functions will be included into the library (not only those reachable from the `@EntryPoint`). However the compiler will _not_ generate the C++-callable wrapper, and we will have to work around.
I'm filing #521 to track that. @irinayat-MS, please let me know if we already have an issue for tracking this. I'll resolve one of them as a duplicate of the other of them and merge the info from both of them into one.

Reply:
And, btw, the block has the following comment: "// The purpose of this block is to keep the Q# compiler from optimizing away other tests when generating QIR", but GitHub's PR review UI doesn't show it by default...